components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1120,6 +1120,12 @@ export class GitpodServerEEImpl extends GitpodServerImpl<GitpodClient, GitpodSer
         return slots[0];
     }
 
+    // Get the current number of "active" slots in a team subscription (count all "assigned" and "unassigned", but not "deactivated" or "cancelled").
+    protected async tsGetActiveSlotQuantity(teamSubscriptionId: string): Promise<number> {
+        const slots = await this.teamSubscriptionDB.findSlotsByTeamSubscriptionId(teamSubscriptionId);
+        return slots.filter(slot => !!slot.cancellationDate).length;
```

Comment:
@jankeromnes just to make sure I understand this correctly...
* "active" means **no cancellation date** is set
* this filter returns slots that actually have the cancellation date set

Reply:
Oh, indeed 🤦 excellent catch, thanks!!